### PR TITLE
Fix CA 2000 Part 2

### PR DESCRIPTION
**Pull request by azabbasi**

### CODE REVIEW

File: iothub/device/src/Transport/HttpClientHelper.cs [hunk 1]
```diff
@@ -63,51 +62,50 @@ IWebProxy proxy
...
             {
-                if (handler == null)
+                if (_httpClientHandler == null)
                 {
-                    handler = new WebRequestHandler();
+                    _httpClientHandler = new WebRequestHandler();
                 }
 
-                handler.ClientCertificates.Add(clientCert);
+                (_httpClientHandler as WebRequestHandler).ClientCertificates.Add(clientCert);
```
Comment (drwill-ms) [inline iothub/device/src/Transport/HttpClientHelper.cs:74]: Why not specify WebRequestHandler as the type on line 41?

File: iothub/device/src/Transport/HttpClientHelper.cs [hunk 2]
```diff
@@ -63,51 +62,50 @@ IWebProxy proxy
...
                 }
 
-                handler.UseProxy = (proxy != null);
-                handler.Proxy = proxy;
+                _httpClientHandler.UseProxy = (proxy != null);
+                _httpClientHandler.Proxy = proxy;
             }
 
-            _httpClientObj = handler != null ? new HttpClient(handler) : new HttpClient();
+            _httpClientObj = _httpClientHandler != null ? new HttpClient(_httpClientHandler) : new HttpClient();
```
Comment (drwill-ms) [inline iothub/device/src/Transport/HttpClientHelper.cs:89]: It can't be null, right?

**Review by drwill-ms — APPROVED** [review]

File: iothub/device/src/Transport/HttpClientHelper.cs (hunk 1 above)
Comment (azabbasi) [reply iothub/device/src/Transport/HttpClientHelper.cs:74]: we use WebRequestHandler only for NET451
for the other frameworks, it's a HttpClientHandler

File: iothub/device/src/Transport/HttpClientHelper.cs [hunk 3]
```diff
@@ -532,6 +530,14 @@ public void Dispose()
                     _httpClientObj = null;
                 }
 
+                // HttpClientHandler that is used to create HttpClient will automatically be disposed when HttpClient is disposed
+                // But in case the client handler didn't end up being used by the HttpClient, we explicitly dispose it here.
+                if (_httpClientHandler != null)
+                {
+                    _httpClientHandler?.Dispose();
```
Comment (barustum) [inline iothub/device/src/Transport/HttpClientHelper.cs:537]: nit: question mark here is redundant since we already check if it's not null

**Review by barustum — APPROVED** [review]

**Review by azabbasi — COMMENTED** [review]
comments disabled.

File: iothub/device/src/Transport/HttpClientHelper.cs (hunk 3 above)
Comment (azabbasi) [reply iothub/device/src/Transport/HttpClientHelper.cs:537]: I will change it in an upcoming PR since I have the gates passed already and this is safe.

**Pull request merged**